### ISSUE TITLE
OSK tray icon click possible regression

### ISSUE THREAD

**Issue by DuMuT6p** [linuxmint/cinnamon#9059]
```
 * Cinnamon version (cinnamon --version) Cinnamon 4.4.5
 * Distribution - Linux Mint 19.3 (upgraded from 19.1)
 * Graphics hardware *and* driver used - Nvidia proprietary driver
 * 32 or 64 bit - 64 bit
 ```
**Issue**
First mouse button click on OSK tray icon displays the right click menu.

**Steps to reproduce**
1. First mouse button click on OSK tray icon 

**Expected behaviour**
It used to hide/show keyboard on Linux Mint 19.2.

**Other information**
"Version 1.6 of libxapp introduces a new system tray solution called XAppStatusIcon." - from "New features" page of Mint 19.3. 
Not sure if the previous behavior was with single or double click, so the problem could be "not possible to double click".

**Comment by icarter09** [comment]
Was able to reproduce this issue. Left click results in the same action as right click on the mouse.

**Comment by mtwebster** [comment]
I think this is actually the way onboard (assuming that's what you're talking about) thinks it should be used.

They're using appindicator, which doesn't actually support left-click activation - it's *middle-click* to 'activate'.  Left and right-click display the same menu.

I suppose we need to decide whether to override that behavior and force the primary button to activate and only right-click shows the menu, to restore expected behavior.  I think we probably should, middle clicking is unnatural :)

**Comment by icarter09** [comment]
@mtwebster you are correct on it being Onboard we are talking about. I agree with your thoughts overriding that behavior.

**Comment by mtwebster** [comment]
Hi, I've been messing around with this, and I worry changing the behavior would cause far more problems than it would fix.  Many programs that use appindicator don't set an 'activate' action (like showing their main window, etc..)  So, if I change left-click to perform 'activate', nothing will happen when the user left-clicks those icons.  Ones that exhibit this that I've found so far are transmission and slack (chat client).

There is somewhat good news, however - in onboard's case, you can go into its preferences, and under "General", set "Status icon provider" to "GtkStatusIcon (MATE)".  This will force onboard to use the old tray applet, but you gain the left-click activation.

I don't know yet if this will be a permanent workaround or not :)  But we may at the very least override that setting in mint so keep desirable behavior by default.

**Comment by DuMuT6p** [comment]
Hello @mtwebster,
I added this as a bug, because it used to behave differently. Didn't know about middle click so I suppose as programs evolve we have to use common behavior for easier maintenance. That seems fine by me. Thank you.
